tests/test_D1_model_components.py

```diff
@@ -0,0 +1,147 @@
+#                                end=pd.to_datetime("2018-12-31 23:00:00"),
+#                                freq="H")
+#     yield solph.EnergySystem(timeindex=time_index)
+
+
+class TestTransformerComponent:
+
+    # @pytest.fixture(autouse=True)
+    # def setup_class(self, get_json, get_model):
+    #     self.dict_values = get_json
+    #     self.model = get_model
+    #     self.transformers = {}
+    #     self.busses = {
+    #         "Fuel bus": solph.Bus(label="Fuel bus"),
+    #         "Electricity bus": solph.Bus(label="Electricity bus")
+    #     }  # todo fixture for the whole model if needed for other classes
+
+    def test_transformer_optimize_multiple_input_busses(self):
+        pass
+
+    def test_transformer_optimize_multiple_output_busses(self):
+        pass
+
+    def test_transformer_optimize_single_busses(self):
+        # dict_asset = self.dict_values["energyConversion"][
+        #     "transformer_optimize_single_busses"]
+        # D1.transformer(model=self.model, dict_asset=dict_asset,
+        #                transformers=self.transformers, busses=self.busses)
+        # check if model was updated with transformer (check inputs and outputs)   model.entities[-1]
+        # check if transformers contains transformer    transformers[dict_asset[label]...
+        pass
+
+    def test_transformer_fix_multiple_input_busses(self):
+        pass
+
+    def test_transformer_fix_multiple_output_busses(self):
+        pass
+
+    def test_transformer_fix_single_busses(self):
+        pass
+
+
+class TestSinkComponent:
+    def test_sink_non_dispatchable_multiple_input_busses(self):
+        pass
+        # check if model was updated with sink (check inputs and outputs)
+        # check if sinks now contains sink
+
+    def test_sink_non_dispatchable_single_input_bus(self):
+        pass
+
+    def test_sink_dispatchable_multiple_input_busses(self):
+        pass
+
+    def test_sink_dispatchable_single_input_bus(self):
+        pass
+
+
+class TestSourceComponent:
+    ## non dispatchable
+    def test_source_non_dispatchable_optimize_multiple_output_busses(self):
+        pass
+
+    def test_source_non_dispatchable_optimize_single_output_bus(self):
+        pass
+
+    def test_source_non_dispatchable_fix_multiple_output_busses(self):
+        pass
+
+    def test_source_non_dispatchable_fix_single_output_bus(self):
+        pass
+
+    ## dispatchable
+    def test_source_dispatchable_optimize_normalized_timeseries_multiple_output_busses(self):
+        pass
+
+    def test_source_dispatchable_optimize_normalized_timeseries_single_output_bus(self):
+        pass
+
+    def test_source_dispatchable_optimize_without_timeseries_multiple_output_busses(self):
+        pass
+
+    def test_source_dispatchable_optimize_without_timeseries_single_output_bus(self):
+        pass
+
+    # note: does the following exist?
+    # def test_source_dispatchable_optimize_timeseries_not_normalized_multiple_output_busses(self):
```

Comment:
```
   # note: does the following exist?
    # def test_source_dispatchable_optimize_timeseries_not_normalized_multiple_output_busses(self):
```

Yes, eg. for fuel for a diesel generator and a chp

Reply:
Okay, so I kept these tests. This functionality seems not to be implemented, yet. I could do then while writing tests.